backend/helpers/helpers.go

```diff
@@ -42,66 +42,62 @@ func DBErrorHandling(error error, w http.ResponseWriter) {
 	}
 }
 
-func GetAllTable[T models.AllTables](w http.ResponseWriter, table []T) {
-	result := config.DB.Find(&table)
-	if result.Error != nil {
-		DBErrorHandling(result.Error, w)
-		return
-	}
-	//go yells at you if you don't error check this, better to be safe
+func JsonWriter(w http.ResponseWriter, table interface{}) {
 	errJson := json.NewEncoder(w).Encode(&table)
 	if errJson != nil {
 		http.Error(w, "500", http.StatusInternalServerError)
-		return
 	}
 }
 
-func GetFirstTable[T models.AllTables, P models.AllTables](w http.ResponseWriter, table T, param P) {
+func GetAllTable[T models.AllTables](w http.ResponseWriter, table []T) (error, []T) {
+	result := config.DB.Omit("Doors").Find(&table)
+	if result.Error != nil {
+		DBErrorHandling(result.Error, w)
+		return result.Error, table
+	}
+	return nil, table
+
+}
+
+func GetFirstTable[T models.AllTables, P models.AllTables](w http.ResponseWriter, table T, param P) (error, T) {
 	result := config.DB.Where(&param).First(&table)
 
 	if result.Error != nil {
 		DBErrorHandling(result.Error, w)
-		return
-	}
-	errJson := json.NewEncoder(w).Encode(&table)
-	if errJson != nil {
-		http.Error(w, "500", http.StatusInternalServerError)
-		return
+		return result.Error, table
 	}
+	return nil, table
+
 }
 
-func CreateNewRecord[T models.AllTables](w http.ResponseWriter, table T, err error) {
+func CreateNewRecord[T models.AllTables](w http.ResponseWriter, table T, err error) (error, T) {
 	if err != nil || !CheckEmptyString(table) {
 		if err != nil {
 			log.Print(err)
 		} else {
 			log.Print("Request contains empty string params")
 		}
 		http.Error(w, "400", http.StatusBadRequest)
-		return
+		return errors.New("400"), table
 	}
 
 	result := config.DB.Create(&table)
 	if result.Error != nil {
 		DBErrorHandling(result.Error, w)
-		return
-	}
-	errJson := json.NewEncoder(w).Encode(table)
-	if errJson != nil {
-		http.Error(w, "500", http.StatusInternalServerError)
-		return
+		return result.Error, table
 	}
+	return nil, table
 }
 
-func DeleteById[T models.AllTables](w http.ResponseWriter, table T, Id string) {
+func DeleteById[T models.AllTables](w http.ResponseWriter, table T, Id string) error {
 	result := config.DB.Unscoped().Delete(&table, Id)
 	if result.Error != nil {
 		DBErrorHandling(result.Error, w)
-		return
+		return result.Error
 	}
 	if result.RowsAffected < 1 {
 		http.Error(w, "404", http.StatusBadRequest)
```

Comment:
can we have more meaningful error messages instead of "404", which is already defined in `http.StatusBadRequest`

Reply:
There's a few places where this can be improved but i'm just going to make this one comment.